crypto/ec/ec_lib.c

```diff
@@ -756,6 +756,11 @@ int EC_POINT_get_affine_coordinates_GFp(const EC_GROUP *group,
               EC_R_INCOMPATIBLE_OBJECTS);
         return 0;
     }
+    if (EC_POINT_is_at_infinity(group, point)) {
```

Comment:
Does returning an error where it didn't previously constitute a change in API?

Reply:
Well, I would dare say that an error was inherently reached anyway at a lower level: in general it is not possible to represent the point-at-infinity in affine coordinates (unless I am mistaken, ping @bbbrumley for a sanity check on the math!)

So I would actually say it's a preemptive bug-fix in case the underlying `get_affine_coordinates` implementation forgot to properly do the checks.

The reason why we need to add it at the wrapper level is because the `evp_test` framework requires to specify reason and function if the expected outcome is an error.

Reply:
My google-fu in my current state of sleep-deprivation brings me to [this Crypto StackExchange question](https://crypto.stackexchange.com/questions/6464/how-to-represent-point-at-infinity-in-affine-coordinate) if I try to substantiate the universality of my clam "it is not possible to represent the point-at-infinity in affine coordinates".

I am still not entirely sure of the universality, but I would say that it holds true for Short-Weierstrass and Montgomery curves, which should be the entirety of what the `EC` submodule supports.

Reply:
@romen Correct, in the context of OpenSSL calling `get_affine_coordinates` for the point-at-infinity doesn't really make any sense. Normally that is for "exporting" a point from the `EC` module. I can't find any real documentation, but I would expect if `EC_POINT_get_affine_coordinates_GFp` returns success, resulting `x` and `y` satisfy the curve equation. And for short curves, that is not possible for the point-at-infinity.

Reply:
Thanks @bbbrumley to double check and for digging those commits out!